backend/src/utils/middleware.ts

```diff
@@ -92,10 +92,35 @@ const userExtractor = catchError(async (req: CustomRequest, res: Response, next:
   }
 });
```

Comment:
I think `userExtractor` can be reduced to this now:

```tsx
const userExtractor = catchError(async (req: CustomRequest, res: Response, next: NextFunction) => {
  if (!req.token) {
    res.status(401).json({ error: 'Invalid User Credentials' });
  } else {
    ifUserExtractor(req, res, next);
  }
});
```

Also, the naming of these functions is a bit confusing for me (could be just me).

Reply:
Good point! Updated! I agree that the name `ifUserExtractor` isn't one of the best. I modified it's name to `getUserFromRequest` and applied your refactoring.